packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -189,31 +189,37 @@ export class W3mModal extends LitElement {
   }
 
   private async onNewAddress(caipAddress?: CaipAddress) {
-    const prevConnected = this.caipAddress
-      ? CoreHelperUtil.getPlainAddress(this.caipAddress)
+    const prevCaipAddress = this.caipAddress
+    const prevConnected = prevCaipAddress
+      ? CoreHelperUtil.getPlainAddress(prevCaipAddress)
       : undefined
     const nextConnected = caipAddress ? CoreHelperUtil.getPlainAddress(caipAddress) : undefined
     const isSameAddress = prevConnected === nextConnected
 
+    this.caipAddress = caipAddress
```

Comment:
Are we sure we want to set this before doing some logics?

Reply:
Might it be better to do it at the end?

Reply:
Yeah it's better to do at the end, but seems like there is some async tasks that's blocking it from having address being assigned earlier. Otherwise in [here](https://github.com/reown-com/appkit/blob/main/packages/scaffold-ui/src/modal/w3m-modal/index.ts#L220-L225) it's going to close the modal.